File: worker/src/enrich/contactout_samples.py
from __future__ import annotations

from typing import Any


def is_contactout_sample_response(data: dict[str, Any]) -> bool:
    """ContactOut returns placeholder data when phone API credits are unavailable."""
    message = str(data.get("message", "")).lower()
    if "sample response" in message:
        return True

    profile = data.get("profile")
    if not isinstance(profile, dict):
        return False

    url = str(profile.get("url", "")).lower()
    if "example-person" in url:
        return True

    for key in ("email", "personal_email", "work_email"):
        values = profile.get(key)
        if not isinstance(values, list):
            continue
        for entry in values:
            if "example.com" in str(entry).lower():
                return True

    for key in ("phone", "phones"):
        values = profile.get(key)
        if not isinstance(values, list):
            continue
        for entry in values:
            text = str(entry).lower()
            if "phone number 1" in text or "+1234567891" in text:
                return True

    return False

File: src/lib/contactout-samples.ts
/** ContactOut returns placeholder data when phone API credits are unavailable. */
export function isContactOutSampleResponse(data: Record<string, unknown>): boolean {
  const message = String(data.message ?? "").toLowerCase();
  if (message.includes("sample response")) return true;

  const profile = (data.profile ?? data.data ?? data) as Record<string, unknown>;
  if (!profile || typeof profile !== "object") return false;

  const url = String(profile.url ?? "").toLowerCase();
  if (url.includes("example-person")) return true;

  for (const key of ["email", "personal_email", "work_email"]) {
    const values = profile[key];
    if (!Array.isArray(values)) continue;
    for (const entry of values) {
      if (String(entry).toLowerCase().includes("example.com")) return true;
    }
  }

  for (const key of ["phone", "phones"]) {
    const values = profile[key];
    if (!Array.isArray(values)) continue;
    for (const entry of values) {
      const text = String(entry).toLowerCase();
      if (text.includes("phone number 1") || text.includes("+1234567891")) {
        return true;
      }
    }
  }

  return false;
}
